storybookjs/storybook
# Core: Fix static dirs targeting same destination

Issue: #16732 

## What I did

The previous implementation caused a race-condition if two assets target the same destination

```
  staticDirs: [
    '../src/assets',
    // reproduction of https://github.com/storybookjs/storybook/issues/16732
    { from: '../src/assets/a', to: 'assets' },
    { from: '../src/assets/b', to: 'assets/b' },
  ],
```

With this change, one staticDir after another gets copied without conflicting in a race condition

## How to test

- Checkout this branch
- Revert my "fix" (there is a separate commit for the reproduction)
- Go to "examples/angular-cli" and run `yarn build-storybook`

The error in the mentioned issue should appear (mkdir file already exists)
